File: web/src/types/log.ts
export interface LogListItem {
  id?: number | string
  created_at: string
  external_model: string
  target_model: string
  status_code: number
  duration_ms: number
  upstream_url: string
  request_snippet?: string | null
  response_snippet?: string | null
  error?: string | null
}

export interface LogDetail {
  id?: number | string
  created_at: string
  external_model: string
  target_model: string
  status_code: number
  duration_ms: number
  upstream_url: string
  request_snippet?: string | null
  response_snippet?: string | null
  error?: string | null
}

export interface FetchLogsParams {
  page: number
  pageSize: number
}

export interface FetchLogsResponse {
  items: LogListItem[]
  total: number
}

File: internal/model/log.go
package model

import "time"

type Log struct {
	ID              uint      `json:"id" gorm:"primaryKey;autoIncrement"`
	CreatedAt       time.Time `json:"created_at"`
	Method          string    `json:"method" gorm:"not null"`
	Path            string    `json:"path" gorm:"not null"`
	ClientIP        string    `json:"client_ip" gorm:"not null"`
	ExternalModel   string    `json:"external_model"`
	TargetModel     string    `json:"target_model"`
	UpstreamURL     string    `json:"upstream_url"`
	StatusCode      int       `json:"status_code" gorm:"not null"`
	DurationMs      int64     `json:"duration_ms" gorm:"not null"`
	RequestSnippet  string    `json:"request_snippet" gorm:"type:text"`
	ResponseSnippet string    `json:"response_snippet" gorm:"type:text"`
	Error           string    `json:"error" gorm:"type:text"`
}

